### ISSUE TITLE
extraterm won't start, KDE Neon 

### ISSUE THREAD

**Issue by cpriest** [sedwards2009/extraterm#282]
I've downloaded the latest .53.1 release (also tried the .53.0 release), both of them show this when starting from a shell:

```A JavaScript error occurred in the main process
Uncaught Exception:
/home/cpriest/extraterm/resources/app/node_modules/@extraterm/extraterm-extension-api:1
../../packages/extraterm-extension-api
^

SyntaxError: Unexpected token '.'
at Module._compile (internal/modules/cjs/loader.js:895:18)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:1004:10)
at Module.load (internal/modules/cjs/loader.js:815:32)
at Module._load (internal/modules/cjs/loader.js:727:14)
at Function.Module._load (electron/js2c/asar.js:769:28)
at Module.require (internal/modules/cjs/loader.js:852:19)
at require (internal/modules/cjs/helpers.js:74:18)
at Object.<anonymous> (/home/cpriest/extraterm/resources/app/extraterm/dist/main_process/Main.js:31:35)
at Module._compile (internal/modules/cjs/loader.js:967:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:1004:10)
```

* I tried it from a fresh unzip 
* I tried it from my home directory, also from .local/bin/extraterm...
* I tried it with just the main folder being named just extraterm

Any idea what's up?

**Comment by cpriest** [comment]
I just tested .52.0 as well and same thing.  I will say I'm fairly new to using Linux as a Desktop Environment, but not to Linux.

If I launch it from Dolphin directly or with the .Desktop file both just try to run and show nothing (I expect because of the same error, just no output of the error)

**Comment by sedwards2009** [comment]
This Linux distribution are you using? Is it possible to try the .deb file?

**Comment by cpriest** [comment]
I'll give it a try, its based on Ubuntu.

**Comment by cpriest** [comment]
Okay, I got something different out of the .deb: `/opt/extraterm/extraterm` from a terminal gets me:
```
2020-09-24 12:48:04.207 INFO [MainConfig #0] Looking for application_paths.json at '/opt/extraterm/application_paths.json'
2020-09-24 12:48:04.210 INFO [main #0] Recording logs to  /home/cpriest/.config/extraterm/extraterm.log
A JavaScript error occurred in the main process
Uncaught Exception:
Error: Command failed: /opt/extraterm/resources/app/extraterm/resources/list-fonts-json-binary/linux-x64/list-fonts-json
at checkExecSyncError (child_process.js:630:11)
at Object.execFileSync (child_process.js:648:15)
at Object.func [as execFileSync] (electron/js2c/asar.js:140:31)
at Object.getAvailableFontsSync (/opt/extraterm/resources/app/extraterm/dist/main_process/FontList.js:17:40)
at getFonts (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:631:42)
at main (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:100:28)
at Object.<anonymous> (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:1083:1)
at Module._compile (internal/modules/cjs/loader.js:967:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:1004:10)
at Module.load (internal/modules/cjs/loader.js:815:32)
```
The log file on line 2 only contains that same line 2.

**Comment by cpriest** [comment]
Hmm, I tried running `/opt/extraterm/resources/app/extraterm/resources/list-fonts-json-binary/linux-x64/list-fonts-json` directly and it segfaulted.  This was running it as my own user.

**Comment by cpriest** [comment]
So I downloaded and built list-fonts-json on my system and it ran fine (output attached).  I copied it over to `/opt/extraterm/resources/app/extraterm/resources/list-fonts-json-binary/linux-x64/list-fonts-json` and re-ran extraterm.  Here's what I got from that run:
```
A JavaScript error occurred in the main process
Uncaught Exception:
SyntaxError: Unexpected token
in JSON at position 17497
at JSON.parse (<anonymous>)
at Object.getAvailableFontsSync (/opt/extraterm/resources/app/extraterm/dist/main_process/FontList.js:18:22)
at getFonts (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:631:42)
at main (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:100:28)
at Object.<anonymous> (/opt/extraterm/resources/app/extraterm/dist/main_process/Main.js:1083:1)
at Module._compile (internal/modules/cjs/loader.js:967:30)
at Object.Module._extensions..js (internal/modules/cjs/loader.js:1004:10)
at Module.load (internal/modules/cjs/loader.js:815:32)
at Module._load (internal/modules/cjs/loader.js:727:14)
at Function.Module._load (electron/js2c/asar.js:769:28)
```

[fonts.json.txt](https://github.com/sedwards2009/extraterm/files/5276327/fonts.json.txt)

**Comment by cpriest** [comment]
I suspected (from the prior error) that my heavily customized zsh shell was causing the problem w/ the JSON output, indeed if I run extraterm from a bash shell (not customized), it ran just fine.

So that solves the last problem.  I did re-install the .deb file and tried it from a bash shell and still got the original failure and running it directly still causes the segfault.

**Comment by sedwards2009** [comment]
Interesting. Try running `ldd /opt/extraterm/resources/app/extraterm/resources/list-fonts-json-binary/linux-x64/list-fonts-json` Maybe there is some library missing on your system.

Are you using xorg or Wayland?

**Comment by cpriest** [comment]
xorg

**Comment by cpriest** [comment]
Any idea of different hex address at the ends of the different `ldd` output matters?  I wouldn't think so, but I don't really develop on Linux at the moment (and nothing serious in C/C++/C# in years).

**Comment by sedwards2009** [comment]
The addresses shouldn't matter. A real problem would be if it couldn't find a matching .so file for something.

I'm having trouble coming up with ideas about this. Have you ever used `strace`?

**Comment by cpriest** [comment]
Yeah, I've used strace before.

**Comment by sedwards2009** [comment]
Try running `list-fonts-json` with `strace` and put the output here and I'll try to figure out where it is crashing.

**Comment by cpriest** [comment]
Alright, I thought it might be useful to have the strace of the version I compiled that does run. Just running a `vimdiff` against the two files (attached) shows a similar path until the segfault; might provide some useful insight.

[list-fonts-json.tar.gz](https://github.com/sedwards2009/extraterm/files/5330105/list-fonts-json.tar.gz)

**Comment by sedwards2009** [comment]
Thanks. Which Linux distribution are you using exactly? Also, can attach your home compiled binary, and run `ldd` on it.

Maybe your fontconfig library is incompatible with the one I've complied against.

**Comment by cpriest** [comment]
Sure, here is the `ldd` of the binary on my machine which is KDE Neon 5.19 which is the KDE build of Ubuntu.

```
linux-vdso.so.1 (0x00007ffd6acee000)
libfontconfig.so.1 => /lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007f84aaefb000)
libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f84aad1a000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f84aacff000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f84aab0d000)
libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007f84aaa4e000)
libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f84aaa20000)
libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007f84aaa15000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f84aa9f2000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f84aa8a3000)
/lib64/ld-linux-x86-64.so.2 (0x00007f84aaf67000)
libpng16.so.16 => /lib/x86_64-linux-gnu/libpng16.so.16 (0x00007f84aa86b000)
libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f84aa84f000)
```
[list-fonts-json.zip](https://github.com/sedwards2009/extraterm/files/5348125/list-fonts-json.zip)

**Comment by sedwards2009** [comment]
Which _exact_ version of Ubuntu are you using?

I'm using an Ubuntu 20.04 base, and things work fine. I want to see if can reproduce the problem locally in a VM.

**Comment by cpriest** [comment]
I'm running the latest KDE neon which is a KDE build based upon Ubuntu 20.04.  KDE just released 5.20 today, after updating I tried a dpkg install again, still the same issue.

The main website is https://neon.kde.org/, in [this blog post](https://blog.neon.kde.org/index.php/2020/10/13/kde-neon-5-20/) today I saw that there is a docker image for running KDE as well, I may try that out and just install extraterm to it and see what I get.

**Comment by sedwards2009** [comment]
I don't think that KDE is likely to make a real difference. So it is likely to be something else. I went through the Linux specific code in `list-fonts-json` and tidied some things up. _Maybe_ that will help.

**Comment by cpriest** [comment]
Any chance you could post an updated binary here and I could try it out?

**Comment by cpriest** [comment]
actually, is there anything about my compile-dir (objects, etc) that might be informative?  I could zip up everything and post it here.

**Comment by sedwards2009** [comment]
@cpriest 
Try this. It is a binary compiled the same way with Debia jessie but with a ton of debug line added. Run it and redirect the output to a file and put it here, and hopefully we can get a better idea as to when exactly it crashes. thanks

[list-fonts-json_debug.zip](https://github.com/sedwards2009/extraterm/files/5440737/list-fonts-json_debug.zip)

**Comment by cpriest** [comment]
sorry, a bit behind on my email, it looks like this binary worked fine.  Here is the output: 
[list-fonts-json-debug.log.zip](https://github.com/sedwards2009/extraterm/files/5447278/list-fonts-json-debug.log.zip)

**Comment by sedwards2009** [comment]
hmmm... working fine wasn't what I was looking for. :-)

Maybe I have some compile time options I can play with.

**Comment by sedwards2009** [comment]
The initial problem reported at the start of this issue looks like the synlink problem described in #286 and fixed in 0.54.0.

The `list-fonts-json` puzzle appears to have been cracked by **matt-e** over in issue #304 !

Thanks for your help!

**Comment by sedwards2009** [comment]
Fixed in v0.55.1.

**Comment by cpriest** [comment]
Hmm, that's interesting because at the start of this you asked me to try the .deb file which I installed w/ dpkg from the command line.  

I may initially have tried dolphin extract but it would have been unusual for me not to have used unzip/tar CLI tool at some point.

I'll check out the next release/55.1 and re-comment here if there are any further problems.  Thanks so much!
